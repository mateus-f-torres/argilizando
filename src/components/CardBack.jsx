//@flow
import * as React from 'react';
import {Link} from 'react-router-dom';

type Props = {
  name: string,
  pitch: string,
  skill: Array<string>,
  tool?: Array<string>,
  lang?: Array<number | string>,
  equip: Array<Array<string>>,
  lockBack: () => void
};

const CardBack = (props: Props) => (
  <figure className="card">
    <h2>{props.name}</h2>
    <p>{props.pitch}</p>
    <div id="hide" className="horizontal-line" />
    <h3>Background Features</h3>
    <ul id="skills" className="multi">
      <h4>Skill Proficiency</h4>
      {props.skill.map((sk, i) => <li key={i}>{sk}</li>)}
    </ul>

    {
      props.tool &&
        <ul id="tools" className="multi">
          <h4>Tool Proficiency</h4>
          {props.tool.map((item, i) => <li key={i}>{item}</li>)}
        </ul>
    }

    {
      props.lang &&
        <ul id="langs" className="multi">
          <h4>Languages</h4>
          <li>{props.lang[1]}</li>
        </ul>
    }

    <ul id="equip" className="dotted-list">
      <h4>Equipment</h4>
      {
        props.equip.map((list, i) => (
          <li key={i}>
            {list.map((item, j) => <span key={j}>{item}<br/></span>)}
          </li>
        ))
      }
    </ul>
    <div className="horizontal-line no-bottom" />
    <div className="lock-box">
      <Link to="/background">
        <button>Back</button>
      </Link>
      <Link to="/">
        <button onClick={props.lockBack}>Lock</button>
      </Link>
    </div>
  </figure>
);

export default CardBack;
