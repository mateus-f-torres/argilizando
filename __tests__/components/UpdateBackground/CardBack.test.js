import React from 'react';
import {mount} from 'enzyme';
import {MemoryRouter} from 'react-router-dom';

import CardBack from 'Components/UpdateBackground/CardBack.js';
import BACKGROUNDS from 'DATA/backgrounds.js';
import {getAsset} from 'Helpers';

describe('<CardBack />', () => {
  test('call lockBack when lock button clicked', () => {
    const mockOnClick = jest.fn();

    const mockBack = getAsset('acolyte', BACKGROUNDS);

    const wrapper = mount(
      <MemoryRouter initialEntries={
        [{pathname: '/background/acolyte', key: 'testKey'}]}>
        <CardBack lockBack={mockOnClick} {...mockBack} />
      </MemoryRouter>
    );

    wrapper.find('[data-test="back-lock"]').simulate('click');
    expect(mockOnClick.mock.calls.length).toBe(1);
  });

  test('if (props.tool) render tools', () => {
    const mockBack = getAsset('charlatan', BACKGROUNDS);

    const wrapper = mount(
      <MemoryRouter initialEntries={
        [{pathname: '/background/charlatan', key: 'testKey'}]}>
        <CardBack {...mockBack} />
      </MemoryRouter>
    );

    expect(wrapper.find('[data-test="back-tools"]').exists()).toBe(true);
  });

  test('if (!props.tool) dont render tools', () => {
    const mockBack = getAsset('heretic', BACKGROUNDS);

    const wrapper = mount(
      <MemoryRouter initialEntries={
        [{pathname: '/background/heretic', key: 'testKey'}]}>
        <CardBack {...mockBack} />
      </MemoryRouter>
    );

    expect(wrapper.find('[data-test="back-tools"]').exists()).toBe(false);
  });

  test('if (props.lang) render langs', () => {
    const mockBack = getAsset('acolyte', BACKGROUNDS);

    const wrapper = mount(
      <MemoryRouter initialEntries={
        [{pathname: '/background/acolyte', key: 'testKey'}]}>
        <CardBack {...mockBack} />
      </MemoryRouter>
    );

    expect(wrapper.find('[data-test="back-langs"]').exists()).toBe(true);
  });

  test('if (!props.tool) dont render langs', () => {
    const mockBack = getAsset('sailor', BACKGROUNDS);

    const wrapper = mount(
      <MemoryRouter initialEntries={
        [{pathname: '/background/sailor', key: 'testKey'}]}>
        <CardBack {...mockBack} />
      </MemoryRouter>
    );

    expect(wrapper.find('[data-test="back-langs"]').exists()).toBe(false);
  });

  test('if (props.tool && props.lang) render both', () => {
    const mockBack = getAsset('noble', BACKGROUNDS);

    const wrapper = mount(
      <MemoryRouter initialEntries={
        [{pathname: '/background/noble', key: 'testKey'}]}>
        <CardBack {...mockBack} />
      </MemoryRouter>
    );

    expect(wrapper.find('[data-test="back-tools"]').exists()).toBe(true);
    expect(wrapper.find('[data-test="back-langs"]').exists()).toBe(true);
  });

  test('match current production CardBack', () => {
    const mockBack = getAsset('acolyte', BACKGROUNDS);

    const wrapper = mount(
      <MemoryRouter initialEntries={
        [{pathname: '/background/acolyte', key: 'testKey'}]}>
        <CardBack {...mockBack} />
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
