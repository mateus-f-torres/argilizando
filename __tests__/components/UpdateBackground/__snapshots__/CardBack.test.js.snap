// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardBack /> match current production CardBack 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
        "pathname": "/background/acolyte",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/background/acolyte",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/background/acolyte",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <CardBack
      _gold={15}
      bonds="I will protect my faiths' temples"
      equip={
        Array [
          Array [
            "holy symbol",
          ],
          Array [
            "prayer book",
            "or prayer wheel",
          ],
          Array [
            "5 sticks of incense",
          ],
          Array [
            "vestments",
          ],
          Array [
            "set of common clothes",
          ],
          Array [
            "belt pouch containing 15gp",
          ],
        ]
      }
      flaws="I am inflexible in my thinking"
      id="acolyte"
      ideals="I always try to help those in need"
      lang={
        Array [
          2,
          "Two of your choice",
        ]
      }
      name="Acolyte"
      personality="I quote sacred texts"
      pitch="You have spent your life in the service 
        of a temple to a specific god or pantheon of gods. 
        You act as an intermediary between the realm of the holy 
        and the mortal world, performing sacred rites and offering sacrifices
        in order to conduct worshipers into the presence of the divine."
      skill={
        Array [
          "Insight",
          "Religion",
        ]
      }
    >
      <figure
        className="card"
      >
        <h2>
          Acolyte
        </h2>
        <p>
          You have spent your life in the service 
        of a temple to a specific god or pantheon of gods. 
        You act as an intermediary between the realm of the holy 
        and the mortal world, performing sacred rites and offering sacrifices
        in order to conduct worshipers into the presence of the divine.
        </p>
        <div
          className="horizontal-line"
          id="hide"
        />
        <h3>
          Background Features
        </h3>
        <ul
          className="multi"
          id="skills"
        >
          <h4>
            Skill Proficiency
          </h4>
          <li
            key="0"
          >
            Insight
          </li>
          <li
            key="1"
          >
            Religion
          </li>
        </ul>
        <ul
          className="multi"
          data-test="back-langs"
          id="langs"
        >
          <h4>
            Languages
          </h4>
          <li>
            Two of your choice
          </li>
        </ul>
        <ul
          className="dotted-list"
          id="equip"
        >
          <h4>
            Equipment
          </h4>
          <li
            key="0"
          >
            <span
              key="0"
            >
              holy symbol
              <br />
            </span>
          </li>
          <li
            key="1"
          >
            <span
              key="0"
            >
              prayer book
              <br />
            </span>
            <span
              key="1"
            >
              or prayer wheel
              <br />
            </span>
          </li>
          <li
            key="2"
          >
            <span
              key="0"
            >
              5 sticks of incense
              <br />
            </span>
          </li>
          <li
            key="3"
          >
            <span
              key="0"
            >
              vestments
              <br />
            </span>
          </li>
          <li
            key="4"
          >
            <span
              key="0"
            >
              set of common clothes
              <br />
            </span>
          </li>
          <li
            key="5"
          >
            <span
              key="0"
            >
              belt pouch containing 15gp
              <br />
            </span>
          </li>
        </ul>
        <div
          className="horizontal-line no-bottom"
        />
        <div
          className="lock-box"
        >
          <Link
            replace={false}
            to="/background"
          >
            <a
              href="/background"
              onClick={[Function]}
            >
              <button>
                Back
              </button>
            </a>
          </Link>
          <Link
            replace={false}
            to="/"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              <button
                data-test="back-lock"
              >
                Lock
              </button>
            </a>
          </Link>
        </div>
      </figure>
    </CardBack>
  </Router>
</MemoryRouter>
`;
